build/commands/lib/util.js

```diff
@@ -252,6 +252,24 @@ const util = {
     fileMap.add([path.join(config.braveCoreDir, 'ui', 'webui', 'resources', 'css', 'text_defaults_md.css'),
                  path.join(config.srcDir, 'ui', 'webui', 'resources', 'css', 'text_defaults_md.css')])
 
+    let explicitSourceFiles = new Set()
+    if (process.platform === 'darwin') {
+      // Set proper mac app icon for channel to chrome/app/theme/mac/app.icns.
+      // Each channel's app icons are stored in brave/app/theme/$channel/app.icns.
+      // With this copying, we don't need to modify chrome/BUILD.gn for this.
+      const iconSource = path.join(braveAppDir, 'theme', 'brave', 'mac', config.channel, 'app.icns')
+      const iconDest = path.join(chromeAppDir, 'theme', 'brave', 'mac', 'app.icns')
+      explicitSourceFiles[iconDest] = iconSource
+
+      // Set proper branding file.
+      let branding_file_name = 'BRANDING'
+      if (config.channel)
```

Comment:
same as above, we should always copy all the files or this is not accurate https://github.com/brave/brave-core/pull/16033#discussion_r1027054377

Reply:
if we remove the channel dependency here then we can update the branding at sync time and only update on build when needed

Reply:
same here. The destination file is a single file, it doesn't depend on the channel, so we can't copy "all the files" into the same destination file..

I have a task to rework this: https://github.com/brave/brave-browser/issues/38309